src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -123,124 +129,94 @@
     private static final String ENCODE_VERSION = "E1";
 
     private static final Object LOCK_OBJ = new Object();
+    
+    private final Context mContext;
+    private final SecureRandom mRandom;
 
+    /**
+     * it is needed for AndroidKeyStore.
+     */
+    private KeyPair mKeyPair;
     private String mBlobVersion;
-
-    private SecretKey mKey = null, mMacKey = null;
-
-    private static SecretKey sSecretKeyFromAndroidKeyStore = null;
+    private SecretKey mKey = null;
+    private SecretKey mMacKey = null;
+    private SecretKey mSecretKeyFromAndroidKeyStore = null;
 
     public StorageHelper(Context ctx) {
         mContext = ctx;
         mRandom = new SecureRandom();
     }
 
     /**
-     * Get Secret Key based on API level to use in encryption. Decryption key
-     * depends on version# since user can migrate to new Android.OS
+     * encrypt text with current key based on API level
      *
-     * @return
-     * @throws NoSuchAlgorithmException
+     * @param clearText Clar text to encrypt. 
+     * @return Encrypted blob.
+     * @throws GeneralSecurityException
+     * @throws UnsupportedEncodingException
      */
-    public synchronized SecretKey loadSecretKeyForAPI() throws IOException, GeneralSecurityException {
-        // Loading key only once for performance. If API is upgraded, it will
-        // restart the device anyway. It will load the correct key for new API.
-        if (mKey != null && mMacKey != null)
-            return mKey;
+    public String encrypt(String clearText)
```

Comment:
Cannot make it package visible, broker is implementing its own cache, but also use the encrypt/decrypt provided in the helper.